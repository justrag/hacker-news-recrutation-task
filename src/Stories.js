import React from 'react';
import { connect } from 'react-redux';
import { ProgressBar, Content } from 'react-mdl';
import {
  getAPIError,
  getStoriesArr,
  getLoaded,
  getLoadingProgress
} from './selectors';
import Story from './Story';

const APIError = () => (
  <Content>
    Sorry, cannot connect to HackerNews. Either you're offline or HackerNews is
    down.
  </Content>
);
const withAPIErrorHOC = Component => ({ apiError, ...rest }) =>
  apiError ? <APIError /> : <Component {...rest} />;

const LoadingProgress = ({ progress }) => (
  <Content>
    <ProgressBar progress={progress} />
  </Content>
);

const withLoadingProgressHOC = Component => ({ loaded, progress, ...rest }) =>
  loaded ? <Component {...rest} /> : <LoadingProgress progress={progress} />;

const Stories = ({ stories }) => (
  <div>{stories.map(story => <Story key={story.id} {...story} />)}</div>
);

const mapStateToProps = state => ({
  apiError: getAPIError(state),
  loaded: getLoaded(state),
  progress: getLoadingProgress(state),
  stories: getStoriesArr(state)
});

export default connect(mapStateToProps)(
  withAPIErrorHOC(withLoadingProgressHOC(Stories))
);
