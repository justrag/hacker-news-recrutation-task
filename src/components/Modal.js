import React from 'react';
import { connect } from 'react-redux';
import { IconButton, Content } from 'react-mdl';
import { getModalHref } from '../selectors/';
import { hideModal } from '../actions/';

const modalStyle = {
  position: 'fixed',
  backgroundColor: '#795548',
  zIndex: 1,
  left: '2%',
  top: '10%',
  width: '95%',
  height: '90%',
  //  overflow: 'auto',
  border: 'thin solid darkgrey'
};
const modalContentStyle = {
  backgroundColor: '#fefefe',
  margin: '15px auto',
  padding: '10px',
  border: '1px solid #888',
  width: '90%',
  height: '90%',
  display: 'block'
};

const closeButtonStyle = {
  float: 'right'
};

export const Modal = ({ href, hideModal }) =>
  href && (
    <Content style={modalStyle}>
      <IconButton
        style={closeButtonStyle}
        onClick={hideModal}
        name="close"
        colored
      />
      <iframe style={modalContentStyle} src={href} />
    </Content>
  );

const mapStateToProps = state => ({ href: getModalHref(state) });

export default connect(mapStateToProps, { hideModal })(Modal);
