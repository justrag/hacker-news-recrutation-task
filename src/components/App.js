import React from 'react';
import { Layout, Header, Footer } from 'react-mdl';
import Modal from './Modal';
import Stories from './Stories';

const layoutStyle = {
  background: 'url(Atari-800xl.jpg) center / cover'
};
//pic from http://bestandworstever.blogspot.com/2012/12/best-atari-system-ever-800xl.html

const titleStyle = { color: '#ddd' };
export const App = () => (
  <div>
    <Layout fixedHeader style={layoutStyle}>
      <Header title={<span style={titleStyle}>HackerNews</span>} />
      <Stories />
      <Modal />
      <Footer size="mini">
        <span>made by rag in 2018</span>
      </Footer>
    </Layout>
  </div>
);

export default App;
