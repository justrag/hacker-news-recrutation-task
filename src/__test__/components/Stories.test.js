import React from 'react';
import { shallow } from 'enzyme';
import { Stories } from '../../components/Stories';

const stories = [
  { id: 42, title: 'The answer' },
  { id: 44, title: 'Old heroes in his blood' }
];

describe('Component: Stories', () => {
  it('renders correctly', () => {
    const test = shallow(<Stories stories={stories} />);
    expect(test).toMatchSnapshot();
  });
});
