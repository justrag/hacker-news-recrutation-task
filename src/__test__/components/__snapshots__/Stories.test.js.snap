// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Stories renders correctly 1`] = `
<div>
  <Connect(Story)
    id={42}
    key="42"
    title="The answer"
  />
  <Connect(Story)
    id={44}
    key="44"
    title="Old heroes in his blood"
  />
</div>
`;
